doc/extensions/zephyr/link-roles.py

```diff
@@ -0,0 +1,40 @@
+# based on http://protips.readthedocs.io/link-roles.html
+
+from __future__ import print_function
+from __future__ import unicode_literals
+import re
+import os
+from docutils import nodes
+from local_util import run_cmd_get_output
+
+
+def get_github_rev():
+    tag = run_cmd_get_output('git describe --exact-match')
+    if len(tag):
+        path = tag
+    else:
+        return 'master'
+
+
+def setup(app):
+    rev = get_github_rev()
+
+    # links to files or folders on the GitHub
+    baseurl = 'https://github.com/zephyrproject-rtos/zephyr'
+    app.add_role('zephyr_file', autolink('{}/blob/{}/%s'.format(baseurl, rev)))
```

Comment:
We need to document these two new roles in our doc guidelines (doc/documentation/doc-guidelines.rst).  Let me know if you'd like me to do that in another PR.

Reply:
can be added later when you do a mass search/replace :)